mantidproject/mantid
# LoadEventNexus change tofmin to be at the lowest event's tof

Since histogramming is inclusive on the lower bound, there is need to shift the lower boundary to the left. The effect of the current implementation is that when the smallest TOF is <1, the workspace histogram started at negative values which makes a mess for log (re)binning.

*There is no associated issue,* but this is tracked with [EWM6796](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=6796)



### To test:

This was originally seen with `SNAP_61408`, but load any event file where the minimum time-of-flight is between zero and one. After using `LoadEventNexus`, view the data and see what the minimum x-value is using the "show data" visualization. Also note that the logs will say something like
```
LoadEventNexus-[Information] Read 1677139 events. Shortest TOF: 0.7 microsec; longest TOF: 16666.6 microsec.
```
before this change, the minimum bin boundary will be -0.3 micorseconds; after this change it will be 0.7 microseconds.



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
